From ddf5cc0b8a81071f58125bee1e86ac214a8cda60 Mon Sep 17 00:00:00 2001
From: Ido Yariv <ido@wizery.com>
Date: Tue, 16 Nov 2010 21:37:11 +0200
Subject: mmc: Fix wakelocks issues in mmc_host_lazy_disable

mmc_host_lazy_disable can get called quite frequently. Each time it is
called, it reschedules a delayed work by calling
mmc_schedule_delayed_work. This function will acquire a wake lock, and
schedule work.

The end result is that the wakelock will be held until there's a
relatively long period in which there is no SDIO traffic at all,
preventing the system from suspending.

Fix this by not acquiring the wakelock, and scheduling work directly.
The wakelock does not seem to be critical here, and can only cause
damage.

Change-Id: Iaa2584f65002b49c96be90744e3d35272e67cdd6
Signed-off-by: Ido Yariv <ido@wizery.com>
Signed-off-by: Vishal Mahaveer <vishalm@ti.com>
---
 drivers/mmc/core/core.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index ed15433..7f73449 100644
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -581,9 +581,8 @@ int mmc_host_lazy_disable(struct mmc_host *host)
 		return 0;
 
 	if (host->disable_delay) {
-		mmc_schedule_delayed_work(&host->disable,
+		return queue_delayed_work(workqueue, &host->disable,
 				msecs_to_jiffies(host->disable_delay));
-		return 0;
 	} else
 		return mmc_host_do_disable(host, 1);
 }
-- 
1.6.0.4

